packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/ClientMethod.java

```diff
@@ -511,6 +511,16 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
                 ClassType.BINARY_DATA.addImportsTo(imports, false);
                 ClassType.HTTP_REQUEST.addImportsTo(imports, false);
             }
+
+            // sync-stack, lro (+ pageable)
+            if (settings.isSyncStackEnabled() && settings.isFluent()) {
+                if ((type == ClientMethodType.PagingSyncSinglePage
+                    && proxyMethod != null
+                    && GenericType.Response(ClassType.BINARY_DATA).equals(proxyMethod.getReturnType().getClientType()))
+                    || type == ClientMethodType.LongRunningBeginSync) {
```

Comment:
nit, maybe put `type == ClientMethodType.LongRunningBeginSync` first, slighter better readability (or put a e.g. `isLroPageable` local var) -- a pity we didn't have better way to check this type

Reply:
updated. Yes, now it's more readable.